### PR TITLE
[REEF-773] Fix typos in README.md of REEF-Tang

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
This PR fixes some typos and broken links.

JIRA:
  [REEF-773](https://issues.apache.org/jira/browse/REEF-773)

Pull Request:
  This closes #507

### CODE REVIEW

File: lang/java/reef-tang/README.md [hunk 1]
```diff
@@ -31,7 +31,7 @@ Tang is our attempt to address these problems.  It consists of a dependency inje
...
+This documentation consists of tutorials that present preferred Tang design patterns.  By structuring your application according to the patterns we suggest throughout the tutorials, you will allow our static analysis framework, Tint ("Tang Lint"), to detect problematic design patterns and high-level configuration problems as part of your build.  These patterns provide the cornerstone for a number of more advanced features, such as interacting with legacy configuration systems, designing for cross-language applications, and multi-tenancy issues, such as secure injections of untrusted application code.  To the best of our knowledge, implementing such tools and addressing these real-world implementation constraints would be difficult, or even impossible, atop competing frameworks.
```
Comment (dongjoon-hyun) [inline lang/java/reef-tang/README.md]: Although Github does not show the difference clearly, `preferred` replaced `prefered`.

File: lang/java/reef-tang/README.md [hunk 2]
```diff
@@ -49,18 +49,18 @@ Tang provides an alternative called `ConfigurationModule`s:
...
+Taken together, these properties greatly simplify dependency injection in distributed environments.  We expect Tang to be used in environments that are dominated by "plugin"-style APIs with many alternative implementations.  Tang cleanly separates concerns over configuration management, dependency injection and object implementations, which hides most of the complexity of dependency injection from plugin implementers.  It also prevents plugin implementations from inadvertently conflicting with each other or their runtime environments.  Such clean semantics are crucial in distributed, heterogeneous environments.
```
Comment (dongjoon-hyun) [inline lang/java/reef-tang/README.md]: Here, `environments` replaced `environements`.

File: lang/java/reef-tang/README.md [hunk 3]
```diff
@@ -430,14 +430,14 @@ Although this works, it has a few disadvantages:
...
+Instead of passing `b` directly into `a`'s constructor, Tang creates an `InjectionFuture<B>` with a single method `public B get()`.  `get()` is guaranteed to throw an exception if called before `A`'s constructor returns, and is guaranteed to return the injected instance of `B` if called after Tang's `Injector.getInstance()` successfully returns.  This allows application designers to break cycles in injections without resorting to mutable fields, or runtime error checking.  `get()` throws a `RuntimeException` that object implementors should ignore and pass up the stack.  Note that calling methods that transitively call `InjectionFuture.get()` within a constructor is fundamentally unsafe.  Such calls have undefined behavior, but Tang makes a best effort attempt to throw an exception when such a call is made.
```
Comment (dongjoon-hyun) [inline lang/java/reef-tang/README.md]: Here, `successfully` replaced `successfuly`.

**Comment by bgchun** [comment]
@dongjoon-hyun Thanks for the fixes. This looks good. I will merge.

**Comment by dongjoon-hyun** [comment]
@bgchun Thank you.

**Pull request closed without merging**